core/lib/storage/src/db.rs

```diff
@@ -274,6 +274,8 @@ pub struct RocksDBOptions {
     /// Timeout to wait for the database to run compaction on stalled writes during startup or
     /// when the corresponding RocksDB error is encountered.
     pub stalled_writes_retries: StalledWritesRetries,
+    /// Number of open files that can be used by the DB. Default is -1, for no limit.
+    pub max_open_files: i32,
```

Comment:
The fact that RocksDB uses `i32` for this param doesn't mean that it makes sense to use it in Rust 🙃 I'd rather use a type that makes sense from the Rust type system perspective, e.g. `Option<NonZeroU32>` (IMO, it would be OK to saturate values over `i32::MAX`, or you could error in the `RocksDB` constructor if such a value is provided.)

Reply:
Erroring out is apparently impossible w/o changing the signature (rocksdb::Error cannot be constructed), so saturating...